test/membrane_caps_audio_raw_test.exs

```diff
@@ -0,0 +1,342 @@
+defmodule Membrane.RawAudioTest do
+  use ExUnit.Case, async: true
+  alias Membrane.Caps.Audio.Raw, as: RawAudio
+
+  @all_formats [
+    :s8,
+    :u8,
+    :s16le,
+    :u16le,
+    :s16be,
+    :u16be,
+    :s24le,
+    :u24le,
+    :s24be,
+    :u24be,
+    :s32le,
+    :u32le,
+    :s32be,
+    :u32be,
+    :f32le,
+    :f32be,
+    :f64le,
+    :f64be
+  ]
+
+  defp format_to_caps(format) do
+    %RawAudio{format: format, channels: 2, sample_rate: 44_100}
+  end
+
+  test "sample_size/1" do
+    sizes = [1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4]
+
+    @all_formats
+    |> Enum.map(&format_to_caps/1)
+    |> Enum.zip(sizes)
+    |> Enum.each(fn {caps, size} ->
+      assert RawAudio.sample_size(caps) == size
```

Comment:
Maybe:
```
expected_sizes = [1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4]
for caps <- Enum.map(@all_formats, &format_to_caps/1), 
    expected_size <- expected_sizes do
    assert RawAudio.sample_size(caps) == size
end
```
This would be resilient for mismatch in lists length.

PS. If you consider this comment as meaningful, there are a few more places where this suggestion aplies.

Reply:
This won't work as `for` will generate a cartesian product of two sets of values:
```elixir
iex(1)> all = [1, 2, 3]
[1, 2, 3]
iex(2)> bs = [:a, :b]           
[:a, :b]
iex(3)> for a <- all, b <- bs do
...(3)> {a, b}
...(3)> end
[{1, :a}, {1, :b}, {2, :a}, {2, :b}, {3, :a}, {3, :b}]
```